server.js

```diff
@@ -49,27 +48,29 @@ app.get("/bookings/search", function (request, response) {
 app.get("/bookings/:id", function (request, response) {
   const bookingId = parseInt(request.params.id);
   let id_booking = bookings.find((booking) => booking.id === bookingId);
-  id_booking ? response.status(200).json(id_booking) : response.status(404).json({message: "id not found" });
+  id_booking
+    ? response.status(200).json(id_booking)
+    : response.status(404).json({ message: "id not found" });
 });
 
 //app delete with id
 app.delete("/bookings/:id", function (request, response) {
   const bookingId = parseInt(request.params.id);
   let toDelete = bookings.find((booking) => booking.id === bookingId);
- if (toDelete) {
+  if (toDelete) {
     let id_booking = bookings.findIndex((booking) => booking.id === bookingId);
     let modifiedArray = bookings.splice(id_booking, 1);
     response.sendStatus(204);
+  } else {
+    response.status(404).json({ message: "id not found" });
   }
-  else { response.status(404).json({ 'message': 'id not found' } )}
-  
 });
 
-
 const searchDate = (date) => {
-  return bookings.filter(booking=> moment(date).isBetween(booking.checkInDate,booking.checkOutDate))
-
-}
+  return bookings.filter((booking) =>
+    moment(date).isBetween(booking.checkInDate, booking.checkOutDate)
+  );
```

Comment:
This does not include the first and last date in the range -  so if the check-in date is 15/11/21 and the check-out date is 17/11/21, only 16/11/21 will return true. I think that the first date of the booking should also be included - it's a matter of opinion whether the check-out date should be included. There's an explanation here on how to include or exclude the first and last dates. https://momentjscom.readthedocs.io/en/latest/moment/05-query/06-is-between/

Reply:
Thank you Karen, yes makes perfect sense, I'll change it asap